model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -1,322 +1,196 @@
+                uprn = sHybrid.uprn,
+                paf = sHybrid.paf.map(_.map(_.toPAFWithFormat)),
+                nag = sHybrid.lpi.map(_.map(_.toNagWithFormat)),
+                underlyingScore = sHybrid.score,//??
+                underlyingMaxScore = maxScore
+              )
+            }
+          ),
+          limit = limit,
+          offset = offset,
+          total = total
+        )
+      ),
+      status = status,
+      errors = errors
+    )
+  }
 }
 
-/**
-  * Contains relevant, to the address request, data
-  *
-  * @param tokens    address decomposed into relevant parts (building number, city, street, etc.)
-  * @param addresses found addresses
-  * @param limit     max number of found addresses
-  * @param offset    offset of found addresses (for pagination)
-  * @param total     total number of found addresses
-  */
-case class AddressBySearchResponse(
+case class Results(
   tokens: Seq[CrfTokenResult],
-  addresses: Seq[AddressResponseAddress],
+  addresses: Option[Seq[AddressInformation]],
   limit: Int,
   offset: Int,
   total: Int
 )
 
-object AddressBySearchResponse {
-  implicit lazy val addressBySearchResponseFormat: Format[AddressBySearchResponse] = Json.format[AddressBySearchResponse]
-  implicit lazy val tokenResultFmt: Format[CrfTokenResult] = Json.format[CrfTokenResult]
-}
-
-
-/**
-  * Contains tokens that build that the address can be splitted onto
-  *
-  * @param uprn           uprn
-  * @param buildingNumber building number
-  * @param postcode       postcode
-  */
-case class AddressTokens(
-  uprn: String,
-  buildingNumber: String,
-  postcode: String
-)
-
-object AddressTokens {
-  implicit lazy val addressTokensFormat: Format[AddressTokens] = Json.format[AddressTokens]
-  /**
-    * Empty tokens (when needed before address tokenization)
-    */
-  val empty = AddressTokens(
-    uprn = "",
-    buildingNumber = "",
-    postcode = ""
-  )
+object Results {
+  implicit lazy val trFmt: Format[CrfTokenResult] = Json.format[CrfTokenResult]
+  implicit lazy val rFmt: Format[Results] = Json.format[Results]
 }
 
-/**
-  * Contains address information retrieved in ES (PAF or NAG)
-  *
-  * @param uprn               uprn
-  * @param formattedAddress   cannonical address form
-  * @param paf                optional, information from Paf index
-  * @param nag                optional, information from Nag index
-  * @param geo                optional, geo information
-  * @param underlyingScore    score from elastic search
-  * @param underlyingMaxScore maximum score in the elastic result
-  */
-case class AddressResponseAddress(
+case class AddressInformation(
   uprn: String,
-  formattedAddress: String,
-  paf: Option[AddressResponsePaf],
-  nag: Option[AddressResponseNag],
-  geo: Option[AddressResponseGeo],
+  paf: Option[Seq[PAFWithFormat]],
+  nag: Option[Seq[NAGWithFormat]],
   underlyingScore: Float,
   underlyingMaxScore: Float
 )
 
-object AddressResponseAddress {
```

Comment:
I moved it to a more OO model

`as an <X> I know how to be transformed to <Y>`